llvm/lib/Analysis/ConstantFolding.cpp

```diff
@@ -3872,6 +3968,26 @@ ConstantFoldStructCall(StringRef Name, Intrinsic::ID IntrinsicID,
       return nullptr;
     return ConstantStruct::get(StTy, SinResult, CosResult);
   }
+  case Intrinsic::vector_deinterleave2: {
+    auto *Vec = dyn_cast<Constant>(Operands[0]);
+    if (!Vec)
+      return nullptr;
+
+    unsigned NumElements =
+        cast<VectorType>(Vec->getType())->getElementCount().getKnownMinValue() /
```

Comment:
If Vec can be scalable here, should we check if it's scalable and bail? Otherwise we're relying on getAggregateElement to return nullptr

Reply:
There really isn't a reason we couldn't support any scalable vector splat today by using Constant::getSplatValue() which will look through the insertelt+shufflevector pattern.